### ISSUE TITLE
Documentation: Update Software Spec (standardize driver feedback)

### ISSUE THREAD

**Issue by Citysurvivor** [FRC2706/2020-2706-Robot-Code#36]
(Updated Feb 6th for clarity:)

The software spec document should be updated:

If multiple programmers use driver feedback, their feedback should be standardized so that all warnings/notifications are consistent, to avoid confusing the drive team. For example, flashing orange lights on the corners = caution, controller vibration = urgent warning (i.e. "motor current cutoff imminent").

Matching feedback to what the drive team prefers would be nice, but not mandatory.

**Comment by georgetzavelas** [comment]
This is good feedback but to make a change to the spec we need to specify exactly what it should be. Having it being interpreted to drive team's preferences is too subjective to meet as a requirement IMHO. We can of course have a requirement that we can allow driver feedback (we need to specify what that is) to be customizable 😄

**Comment by nicojopr** [comment]
For the general design documentation task:
For example of previous software spec, see  last year's  write ups https://drive.google.com/drive/folders/1t3DF-qLRp0zoJyNotzSUO6AjYBazRQiq?usp=sharing

We propose to have each person update the wiki this year for their area of contribution, then we can pull info together. In the meantime, good to start a skeleton doc for 2020 Software in https://drive.google.com/drive/folders/1IvX7M4hOTm0dBTsD9Jj7GpSdslO8-dWY

Link to folder for writeups with template [https://docs.google.com/document/d/1yA21zNQh31z1THF45gdJ3XXEMdJyRvJdj1Vx6dU6RnM/edit](url)

**Comment by Citysurvivor** [comment]
> This is good feedback but to make a change to the spec we need to specify exactly what it should be. Having it being interpreted to drive team's preferences is too subjective to meet as a requirement IMHO. We can of course have a requirement that we can allow driver feedback (we need to specify what that is) to be customizable 😄

True, although now that I reread my issue, I realize that I didn't mean to say that. Updated my post for clarity.

**Comment by nicojopr** [comment]
Link to template to use for each software area documentation https://docs.google.com/document/d/1yA21zNQh31z1THF45gdJ3XXEMdJyRvJdj1Vx6dU6RnM/edit#heading=h.3as4poj
